src/app/server/Dnssd.cpp

```diff
@@ -78,6 +63,26 @@ void OnPlatformEventWrapper(const DeviceLayer::ChipDeviceEvent * event, intptr_t
 
 constexpr System::Clock::Timestamp DnssdServer::kTimeoutCleared;
 
+DnssdServer::DnssdServer()
+{
+    mFabricTable = &Server::GetInstance().GetFabricTable();
+}
```

Comment:
This should NOT be in the constructor, but rather post-consctruction init. Otherwise a static isntance of a DnssdServer could never exist unless `Server` class was initialized. Please make an API method to inject the mFabricTable, and let the caller figure-out where to pass it.

Reply:
Good catch. Yes, agreed that we should not reach out to static instances from a constructor. It works in this instance only because a) Server gets constructed on first use and b) the fabric table member in that class is also constructed in tandem at the time the server gets constructed.

I already have a `SetFabricTable` API on it that the controller invokes - will just make the `Server` call into that as well.